In [21]:
import numpy as np
import sympy as sp

__1.__ Найдите собственные векторы и собственные значения для линейного оператора, заданного матрицей:

$A=\begin{pmatrix}
-1 & -6\\
2 & 6
\end{pmatrix}.$

__Решение__

Вычтем «лямбду» из всех чисел главной диагонали матрицы

$\left|
\begin{array}{cc}
 -1-\lambda  & -6 \\
 2 & 6-\lambda  \\
\end{array}
\right|=0$

$(-\lambda -1) (-\lambda +6) + 12 =\lambda ^2-5 \lambda -6 + 12 = (\lambda -3) (\lambda -2)$

$\Rightarrow
\begin{cases}
\lambda=3\\
\lambda=2\\
\end{cases}
$

$\begin{pmatrix}
-1 & -6\\
2 & 6
\end{pmatrix}
\begin{pmatrix}
x_{1}\\
x_{2}
\end{pmatrix}=
\lambda
\begin{pmatrix}
x_{1}\\
x_{2}
\end{pmatrix}$

$\lambda = 2:$

$\begin{cases}
-x_{1}-6x_{2} = 2x_{1}\\
2x_{1}+6x_{2} = 2x_{2}
\end{cases}\Rightarrow
\begin{cases}
3x_{1} = -6x_{2}\\
2x_{2} = -x_{1}
\end{cases}\Rightarrow
\begin{cases}
x_{1} = -2\\
x_{2} = 1
\end{cases}$


$\lambda = 3:$

$\begin{cases}
-x_{1}-6x_{2} = 3x_{1}\\
2x_{1}+6x_{2} = 3x_{2}
\end{cases}\Rightarrow
\begin{cases}
4x_{1} = -6x_{2}\\
2x_{1} = -3x_{2}
\end{cases}\Rightarrow
\begin{cases}
x_{1} = -\frac{3}{2}\\
x_{2} = 1
\end{cases}
$

In [41]:
A = np.array([[-1, -6], [2, 6]])
L, Vn = np.linalg.eig(A)

As = sp.Matrix([[-1, -6], [2, 6]])
Vs = [list(i[2][0]) for i in As.eigenvects()]
print(
    f"Собственные значения: {L} \nНормированные собственные векторы:\nотвечающий первому собственному значению {Vn[:, 0]},\nотвечающий второму собственному значению {Vn[:, 1]}\nНенормированные собственные векторы:\nотвечающий первому собственному значению {Vs[0]},\nотвечающий второму собственному значению {Vs[1]}"
)

Собственные значения: [2. 3.] 
Нормированные собственные векторы:
отвечающий первому собственному значению [-0.89442719  0.4472136 ],
отвечающий второму собственному значению [ 0.83205029 -0.5547002 ]
Ненормированные собственные векторы:
отвечающий первому собственному значению [-2, 1],
отвечающий второму собственному значению [-3/2, 1]


Собственными векторами являются также и все векторы вида
$c_1e_1$, $с_2е_2$, где $c_1-c_2$ — произвольные действительные числа не равные $0$.

__Ответ.__ $\lambda_1 = 2$, $e_1 = c_1(-0.89442719,  0.4472136)$; $\lambda_2 = 3$, $e_2 = c_2(0.83205029, -0.5547002)$, $c_1, c_2 \in \mathbb{R} \backslash \{0\}$.

__2.__ Дан оператор поворота на 180 градусов, задаваемый матрицей:

$A=\begin{pmatrix}
-1 & 0\\
0 & -1
\end{pmatrix}.$

Покажите, что __любой__ вектор считается для него собственным.

__Решение__

Характеристическое уравнение $(-1-\lambda)^2$ с кратными собственными числами $\lambda_{1, 2} = -1$

Найдем собственные векторы:

Подставив $\lambda_{1, 2} = -1$ в определитель и запишем однородную систему линейных уравнений:

$\begin{cases}
0 \cdot x+0\cdot y= 0 \\
0 \cdot x+0\cdot y = 0
\end{cases},$

Системе удовлетворяют координаты любого вектора. Таким образом, любой ненулевой вектор этого преобразования является собственным.

In [15]:
A = np.array([[-1, 0], [0, -1]])
L, V = np.linalg.eig(A)
print(
    f"Собственные значения: {L} \nСобственный вектор, отвечающий первому собственному значению {V[:, 0]},\nСобственный вектор, отвечающий второму собственному значению {V[:, 1]}"
)

Собственные значения: [-1. -1.] 
Собственный вектор, отвечающий первому собственному значению [1. 0.],
Собственный вектор, отвечающий второму собственному значению [0. 1.]


__3.__ Пусть линейный оператор задан матрицей:

$A=\begin{pmatrix}
1 & 1\\
-1 & 3
\end{pmatrix}.$

Установите, считается ли вектор $x=(1,1)$ собственным вектором этого линейного оператора.

__Решение__

Предположим, что вектор $x$ считается собственным вектором заданного линейного оператора, тогда должно существовать некоторое вещественное число $\lambda$, при котором:

$\begin{pmatrix}
1 & 1\\
-1 & 3
\end{pmatrix}
\begin{pmatrix}
1\\
1
\end{pmatrix}=
\lambda
\begin{pmatrix}
1\\
1
\end{pmatrix}.
$

Из этого следует:

$\begin{cases}
1+1=\lambda \\
-1+3=\lambda
\end{cases}
\Rightarrow
\begin{cases}
\lambda=2\\
\lambda=2
\end{cases}.$

Следовательно, вектор $x=(1,1)$ считается собственным вектором линейного оператора, заданного матрицей $A$.

Собственными векторами являются также и все векторы вида
$c_1e_1$, $с_2е_2$, где $c_1-c_2$ — произвольные действительные числа не равные $0$.

In [43]:
A = np.array([[1, 1], [-1, 3]])
x = np.array([1, 1])
L = A @ x
np.divide(L, x)

array([2., 2.])

In [54]:
As = sp.Matrix([[1, 1], [-1, 3]])
Vs = [list(i[2][0]) for i in As.eigenvects()]

print(
    f"Собственное значение: {list(As.eigenvals().keys())} \nСобственный вектор {Vs[0]}"
)

Собственное значение: [2] 
Собственный вектор [1, 1]


**4.** Пусть линейный оператор задан матрицей:

$A=\begin{pmatrix} 0 & 3 & 0\\ 3 & 0 & 0\\ 0 & 0 & 3 \end{pmatrix}.$

Установите, считается ли вектор $x=(3, -3, -4)$ собственным вектором этого линейного оператора.

__Решение__

$\begin{pmatrix}
0 & 3 & 0\\
3 & 0 & 0\\
0 & 0 & 3
\end{pmatrix}
\begin{pmatrix}
3\\
-3\\
-4
\end{pmatrix}=
\lambda
\begin{pmatrix}
3\\
-3\\
-4
\end{pmatrix}$

Вычтем «лямбду» из всех чисел главной диагонали матрицы $A$ и составим её характеристическое уравнение:

$
\left|
\begin{array}{ccc}
0-\lambda & 3 & 0\\
3 & 0-\lambda & 0\\
0 & 0 & 3-\lambda
\end{array}
\right| = 0
$

$
\Delta = 0 \left|
\begin{array}{cc}
 3 & 0 \\
 -\lambda  & 0 \\
\end{array}
\right| +0 \left|
\begin{array}{cc}
 -\lambda  & 0 \\
 3 & 0 \\
\end{array}
\right| +(-\lambda +3) \left|
\begin{array}{cc}
 -\lambda  & 3 \\
 3 & -\lambda  \\
\end{array}
\right|\text{= }(3-\lambda ) \left|
\begin{array}{cc}
 -\lambda  & 3 \\
 3 & -\lambda  \\
\end{array}
\right|=(3-\lambda ) (-\lambda  (-\lambda )-3\cdot 3)=(-\lambda +3) \left(\lambda ^2-9\right)=\fbox{$(3-\lambda ) \left(-9+\lambda ^2\right)$}=-(\lambda -3)^2 (\lambda +3)
$

$\Rightarrow
\begin{cases}
\lambda=3\\
\lambda=-3\\
\end{cases}
$

Выясним, является ли вектор $x=(3, -3, -4)$ собственным вектором этого линейного оператора.

$\Rightarrow
\begin{cases}
3\lambda=-9\\
-3\lambda=9\\
-4\lambda=-12
\end{cases}
$ $\Rightarrow
\begin{cases}
\lambda=-3\\
\lambda=-3\\
\lambda=3
\end{cases}$

Таким образом, вектор $x=(3, -3, -4)$ не является собственным вектором линейного оператора, заданного матрицей $A$


In [57]:
As = sp.Matrix([[0, 3, 0], [3, 0, 0], [0, 0, 3]])
Vs = [list(i[2][0]) for i in As.eigenvects()]

print(f"Собственное значение: {list(As.eigenvals().keys())} \nСобственные векторы {Vs}")

Собственное значение: [-3, 3] 
Собственные векторы [[-1, 1, 0], [1, 1, 0]]
